### ISSUE TITLE
update autogen.sh to include automake 1.16

### ISSUE THREAD

**Issue by ellelstone** [mypaint/libmypaint#126]
OpenSUSE Tumbleweed now provides automake 1.16, but autogen.sh reports there is no version of automake greater than 1.13. 

Looking at the code in autogen.sh, it's obvious where to add lines to include a check for 1.16, so I was able to compile libmypaint without any problems. But it would be nice to make the change in the version of autogen.sh that downloads from here, so people running Tumbleweed don't run into this problem.

**Comment by Jehan** [comment]
libmypaint master branch already tests for automake 1.16 (since commit 40d9077a80be13942476f164bddfabe842ab2a45 back from March) so I assume you are talking about libmypaint 1 (and more accurately 1.3.0, last stable release).

I agree this is something which should be backported (as other things probably), but right now we only have a tag "v1.3.0", and in particular no branches. I am unsure if main libmypaint developers are planning any bugfix release v1.x. Anyone to shed some light on the release plans for libmypaint?

**Comment by ellelstone** [comment]
> libmypaint master branch already tests for automake 1.16 (since commit 40d9077 back from March) so I assume you are talking about libmypaint 1 (and more accurately 1.3.0, last stable release).

Hi Jehan, 

I was building gimp git master on an opensuse box (following my own tutorial, which clearly needs to be updated: https://ninedegreesbelow.com/photography/build-gimp-in-prefix-for-artists.html). I cloned this:

git clone git://github.com/mypaint/libmypaint.git

and then switched to the 1.3 branch:
git checkout v1.3.0

I had the same issue with the mypaint brushes:
git clone https://github.com/Jehan/mypaint-brushes.git
git checkout v1.3.0

**Comment by Jehan** [comment]
> and then switched to the 1.3 branch

libmypaint has no 1.3 branch but only a tag, which is the problem I was talking about. Making a branch is easy, but I don't want to take such decision alone since it only makes sense if the main developers of the project decide to make new 1.x releases.
@briend @achadwick What are the future plans with libmypaint?

> I had the same issue with the mypaint brushes

MyPaint brushes repo have a branch. You can check it out instead of the tag: `git checkout v1.3.x`.
It already contains the fix.

**Comment by briend** [comment]
I don't know a whole lot about distribution and packaging issues.  I'm just here to break the ABI, mostly :-)  I'm not sure how @achadwick feels about this but I would love if @Jehan felt free to make changes as you see fit.  A new branch sounds pretty reasonable to me

**Comment by Jehan** [comment]
Ok I'll proceed forward then unless @achadwick wishes me not to.

Basically in the 1.x branch, we should be able to backport at least bug fixes or anything boring which does not break ABI/API. :P

**Comment by briend** [comment]
That would be exciting... we could backport just about everything except that silly viewzoom/rotation stuff what's-his-name added ;-)

**Comment by Jehan** [comment]
Ok! So I created and pushed the branch "libmypaint-v1". I stopped it at commit d842adc06cae8121abee1f2237c3099f17d23741 (back from 2017, March!) because the next commit already breaks some API.

So I propose we work from there. I haven't backported yet the commit from the autogen.sh change which pertains to the current report because we should try to backport things in the right order (less risk to get merge conflicts).
You probably all know, but a good comment to backport is: `git cherry-pick -x <commit-hash>` (the `-x` option adds a small text in the git comment with the original commit hash from master. Very useful if needed for debugging and keeping history).
